### PR TITLE
EFI random seed post #25319 review fixups

### PR DESCRIPTION
**Pull request by zx2c4**
This addresses some post-review items from #25319, in three commits.

### CODE REVIEW

**Comment by zx2c4** [comment]
CC @poettering

File: src/boot/efi/util.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ __attribute__((noreturn)) extern void __assert_cl_failure__(void) __attribute__(
 __attribute__((noreturn)) extern void __assert_cl_failure__(void);
 #endif
 /* assert_cl generates a later-stage compile-time assertion when constant folding occurs. */
-#define assert_cl(condition) if (!(condition)) __assert_cl_failure__()
+#define assert_cl(condition) do { if (!(condition)) __assert_cl_failure__(); } while (0)
```
Comment (medhefgo) [inline src/boot/efi/util.h]: Let's use statement expressions for new code macros: `({})`

**Comment by poettering** [comment]
thank you! lgtm

File: src/boot/efi/util.h (hunk 1 above)
Comment (zx2c4) [reply src/boot/efi/util.h]: Will do.

**Comment by zx2c4** [comment]
Okay changes pushed, should be good to go.

**Comment by zx2c4** [comment]
@bluca An Ubuntu test failed, but this seems like a flake rather than something real?

**Review by bluca — APPROVED** [review]
yeah it's a daemon-reload during runtime, don't see how it could be related to these changes

**Pull request merged**